### DISCUSSION TITLE
Provide unique non-clashed IDs for `RequestContext`

### DISCUSSION THREAD

**Discussion by derevnjuk** [mikro-orm/mikro-orm#4180] [Ideas]
### Description
The current implementation of `RequestContext` holds non-unique IDs over the nodes and runs, which can be a bit tricky to use as an indication of the local transaction since `Knex` does not expose `trxid` as well.

### Possible sollution
I would like to suggest using [UUID](https://github.com/uuidjs/uuid) or [nanoid](https://github.com/ai/nanoid) instead of just a static counter to generate unique IDs for `RequestContext`.

```diff
@@ -1,5 +1,6 @@
 import { AsyncLocalStorage } from 'async_hooks';
 import type { EntityManager } from '../EntityManager';
+import { v4 } from 'uuid';
 
 /**
  * Uses `AsyncLocalStorage` to create async context that holds current EM fork.
@@ -7,8 +8,7 @@
 export class RequestContext {
 
   private static storage = new AsyncLocalStorage<RequestContext>();
-  private static counter = 1;
-  readonly id = RequestContext.counter++;
+  readonly id = v4();
 
   constructor(readonly map: Map<string, EntityManager>) { }
 
```

**Comment by B4nan** [comment]
> which can be a bit tricky to use as an indication of the local transaction

Why would this matter? The whole point of the ID is just for debugging purposes, nothing more, it has no real meaning for runtime.

edit: And when I say debugging purposes, I mean debugging request context issues locally, this information is not meant to be displayed in any app logs.

**Reply by derevnjuk** [reply]
While it's true that the ID in the `RequestContext` may not have a direct impact on runtime, having a unique identifier can be useful in many ways beyond just debugging. The `RequestContext` defines a "unit of work" that is supposed to be done at once. It is logical, that users should be able to distinguish them. For instance, it can be used alongside with the `beforeTransactionCommit` to persist and dispatch the integration events.

**Reply by B4nan** [reply]
>  The RequestContext defines a "unit of work" that is supposed to be done at once. 

The `RequestContext` helper is rather an implementation detail, it's opt-in in the end. What defines the context is the underlying entity manager instance - and you can have many of them created from one request context. Those also have their own id, again just incrementing integer.

> It is logical, that users should be able to distinguish them. 

It will be unique in the given app instance, why would you care about uniqueness outside of it?

I wanted to have a simple number there for two reasons:
- its easy to debug, much simpler to read and compare in your head
- it literally costs nothing to generate, unlike UUID (although nowadays there are faster alternatives too)
- it would require additional dependency, we don't use UUIDs anywhere explicitly

Maybe it would be best to just make this configurable.
